algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+                    visualization (default: 0)""")
+parser.add_argument('--vizdoom', dest='vizdoom', action='store_true',
+                    help='Run vizdoom env instead with name below instead of ai2thor and atari')
+parser.add_argument('--scenario-path', type=str, default="vizdoom_maps/room.wad",
+                    help="""Doom scenario file to load
+                    (default: maps/room.wad)""")
+parser.add_argument('--interactive', type=int, default=0,
+                    help="""Interactive mode enables human to play
+                    (default: 0)""")
+parser.add_argument('--all-instr-file', type=str,
+                    default="vizdoom_data/instructions_all.json",
+                    help="""All instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_all.json)""")
+parser.add_argument('--train-instr-file', type=str,
+                    default="vizdoom_data/instructions_train.json",
+                    help="""Train instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_train.json)""")
+parser.add_argument('--test-instr-file', type=str,
+                    default="vizdoom_data/instructions_test.json",
+                    help="""Test instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_test.json)""")
+parser.add_argument('--object-size-file', type=str,
+                    default="vizdoom_data/object_sizes.txt",
+                    help='Object size file path relative to a3c folder '
+                         '(default: data/object_sizes.txt)')
+parser.add_argument('-e', '--evaluate', type=int, default=0,
+                    help="""0:Train, 1:Evaluate MultiTask Generalization
+                    2:Evaluate Zero-shot Generalization (default: 0)
+                    async must be on and will only run test function""")
 
 if __name__ == '__main__':
-    os.environ['OMP_NUM_THREADS'] = '1'
     os.environ['CUDA_VISIBLE_DEVICES'] = ""
 
+    # todo add to README.md more info
+
     args = parser.parse_args()
+    # set to 0 so that checkpoint resume can overwrite if necessary
+    args.episode_number = 0
+    args.total_length = 0
+    args.num_backprops = 0
+    if not args.experiment_id:
+        args.experiment_id = datetime.datetime.now().strftime("%Y-%m-%d-") \
+                                                                     + str(uuid.uuid4())
+    args.experiment_path = os.path.join(os.path.abspath(os.path.join(os.path.dirname(__file__),
+                           '..', '..', 'experiments', args.experiment_id)))
+    args.checkpoint_path = os.path.join(args.experiment_path, 'checkpoints')
+    args.tensorboard_path = os.path.join(args.experiment_path, 'tensorboard_logs')
 
-    torch.manual_seed(args.seed)
+    # Create environment from choices of Atari, ViZDoom and our ai2thor wrapper with tasks
     if args.atari:
```

Comment:
As mentioned above, don't use 3 arguments for 3 different environments. args.environment or args.game and check the string you have there "vizdoom", "space_invaders" or whatever.

Reply:
I did this. atari, vizdoom and ai2thor. the default atari is pong so you need two arguments to decide which atari env which is fine to me